src/components/Dropdown/index.jsx

```diff
@@ -0,0 +1,108 @@
+import styles from './styles.module.css';
+
+export default function Dropdown({ optionsList, setOption }) {
+  const [selected, setSelected] = useState(optionsList[0]);
+
+  const changeHandler = (val) => {
+    setSelected(val);
+    setOption(val);
+  };
+
+  return (
+    <div className={styles.container}>
+      <Listbox value={selected} onChange={changeHandler}>
+        <div className={cls(
+          styles.innerContainer,
+          margin.mt1,
+        )}
+        >
+          <Listbox.Button className={cls(
+            styles.listButton,
+            padding.py2,
+            typography.textSm,
+            colors.textSlate900,
+            borders.roundedLg,
+          )}
+          >
+            <span className={styles.buttonSpan}>{selected.title}</span>
+            <span className={cls(
+              styles.buttonIcon,
+              padding.pr2,
+              flex.flex,
+              layout.itemsCenter,
+            )}
+            >
+              <ChevronUpDownIcon
+                className={styles.chevronIcon}
+                aria-hidden="true"
+              />
+            </span>
+          </Listbox.Button>
+          <Transition
+            as={Fragment}
+            leave="transition ease-in duration-100"
+            leaveFrom="opacity-100"
+            leaveTo="opacity-0"
+          >
+            <Listbox.Options className={cls(
+              styles.listOptions,
+              typography.textSm,
+              margin.mt1,
+              borders.roundedMd,
+            )}
+            >
+              {optionsList.map((illo) => (
+                <Listbox.Option
+                  key={illo.title + illo.path}
+                  className={({ active }) => `${
+                    active ? cls(
+                      styles.option,
+                      styles.optionActive,
+                      colors.bgSlate600,
+                      padding.py2,
+                    ) : cls(
+                      styles.option,
+                      colors.textSlate900,
+                      padding.py2,
+                    )
+                  }`}
+                  value={illo}
+                >
+                  {({ selected }) => (
+                    <>
+                      <span
+                        className={`${
+                          selected
+                            ? cls(
+                              styles.optionText,
+                              typography.fontBold,
+                            )
+                            : cls(
+                              styles.optionText,
+                              typography.fontNormal,
+                            )
+                        }`}
+                      >
+                        {illo.title}
+                      </span>
+                    </>
+                  )}
+                </Listbox.Option>
+              ))}
+            </Listbox.Options>
+          </Transition>
+        </div>
+      </Listbox>
+    </div>
```

Comment:
This file is getting a little long. It's not wrong, but let's chat when I get back about how we could possibly make it smaller.